.github/workflows/chain-spec-snapshot-build.yml

```diff
@@ -14,7 +14,7 @@ on:
 jobs:
   chains-spec:
     runs-on: ${{ fromJson(github.repository_owner == 'autonomys' &&
-      '"runs-on=${{ github.run_id }}/runner=self-hosted-ubuntu-22.04-x86-64"' || '"ubuntu-22.04"') }}
+      '"runs-on=${{ github.run_id }}-${{ github.run_attempt }}/runner=self-hosted-ubuntu-22.04-x86-64/spot=false"' || '"ubuntu-22.04"') }}
```

Comment:
Why is `github.run_attempt` use in here?

Reply:
If we just use the run-id, then different attempts can be scheduled on the same runner. This disables the spot instance interruption protection on job re-runs:
https://github.com/runs-on/runs-on/issues/337#issuecomment-3109618827

Probably isn't strictly needed in this file with `spot=false`, but I did a search and replace for consistency (and to avoid other similar issues).

Reply:
Such an obscure behavior. I really consider this to be a bug.